presto-main/src/main/java/com/facebook/presto/execution/DataDefinitionExecution.java

```diff
@@ -132,8 +131,8 @@ public void onFailure(Throwable throwable)
         }
         catch (Throwable e) {
             fail(e);
-            if (!(e instanceof RuntimeException)) {
-                throw Throwables.propagate(e);
+            if (e instanceof Error) {
+                throw e;
```

Comment:
I don't understand the reasoning behind selective swallowing of throwable here. In addition, I think swallowing throwables deserves a comment.

Reply:
This is the same logic as `SqlQueryExecution`. I'll change the code to be identical. I assume the reasoning is that a `RuntimeException` is a user error, which we've already handled (by failing the query), but an `Error` is something wrong with the engine that we want to rethrow.